pkg/api/v1/types_kubelet.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/resource"
+)
+
+// Container reference contains enough information to uniquely identify a container
+type ContainerReference struct {
+	// The absolute name of the container. This is unique on the machine.
+	Name string `json:"name"`
+
+	// Other names by which the container is known within a certain namespace.
+	// This is unique within that namespace.
+	Aliases []string `json:"aliases,omitempty"`
+
+	// Namespace under which the aliases of a container are unique.
+	// An example of a namespace is "docker" for Docker containers.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type NodeStats struct {
+	TypeMeta
+	ObjectMeta // should have node name
+	Containers []struct {
+		ContainerReference
+		Stats []*ContainerStats
+	}
+}
+
+type ContainerStats struct {
+	Time time.Time `json:"time"`
+
+	// In cores
+	Cpu resource.Quantity
+	// In bytes
+	Memory resource.Quantity
```

Comment:
We need detailed memory stats here. I think we can re-use most of cadvisor's API here. Some metrics like cpu shares cannot be exposed as-is. The rest we can.